src/main/java/io/micronaut/build/MicronautQualityChecksAggregatorPlugin.java

```diff
@@ -30,18 +30,30 @@ public void apply(final Project rootProject) {
 
 
     private void configureSonar(final Project rootProject) {
-        rootProject.getPluginManager().apply(SonarQubePlugin.class);
-        final SonarQubeExtension sonarQubeExtension = rootProject.getExtensions().findByType(SonarQubeExtension.class);
-        final String githubSlug = (String) rootProject.findProperty("githubSlug");
-        if (sonarQubeExtension != null) {
-            sonarQubeExtension.properties(p -> {
-                if (githubSlug != null) {
-                    p.property("sonar.projectKey", githubSlug.replaceAll("/", "_"));
-                    p.property("sonar.organization", "micronaut-projects");
-                    p.property("sonar.host.url", "https://sonarcloud.io");
-                    p.property("sonar.java.source", "8");
-                }
-            });
+        if (System.getenv("SONAR_TOKEN") != null) {
```

Comment:
:x: I think this conditional is going to break caching, or I should say, if CI builds have the `SONAR_TOKEN` and local builds do not, then the script classpath will be different for CI builds and local builds, which means local builds won't be able to fetch from the cache. So ideally, the plugin should always be applied and the conditional should be moved within for the configuration. But I don't know if the sonar plugin can be applied without configuration without breaking builds.

Reply:
Hmm... I am not sure if the script classpath will be different. Does the `SonarQubePlugin` do something with the script classpath? Since it seems to be already on the classpath when it is applied here, so only applying it wouldn't change anything, right?

Reply:
Ah that's the piece I was missing, you're right. Since we already put the plugin on classpath via dependencies, it's different from a build script applying a plugin conditionally. So we're good, thanks for the insights!